### DISCUSSION TITLE
Getting UMAT state variable vector as an AuxVariable

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#25546] [Q&A General]
I thought I might be able to accomplish this using `VectorMaterialRealVectorValueAux`, but I get an error about the "state_var" material property created by the UMAT material is of type `libMesh::VectorValue<double>` and not `std::vector<double>`. Is there a different AuxKernel I should use?

**Comment by GiudGiud** [comment]
so `std::vector<double>` is going to be the type for an array material property
RealVectorValue is a 3-sized vector (or libmesh_dim-sized)

so instead of `std::vector<double>` you should look at returning RealVectorValue?
what type is your material property?

**Reply by jessecarterMOOSE** [reply]
The material property ( defined here https://github.com/idaholab/moose/blob/81a54ccefe510af7372d5b54e0e7b856c1d3960f/modules/tensor_mechanics/include/materials/abaqus/AbaqusUMATStress.h#L214) gets an arbitrary size.

**Reply by GiudGiud** [reply]
Then you must use array auxkernels for output.

**Reply by jessecarterMOOSE** [reply]
Does moose already have one that would work in this case? I'd like to have the whole array in one auxkernel because the state variable list could be 100 entries long.

**Reply by GiudGiud** [reply]
does this work?
https://mooseframework.inl.gov/source/auxkernels/MaterialStdVectorAux.html

**Reply by jessecarterMOOSE** [reply]
That's not an array kernel, right? Just pulls one component of the vector.

**Reply by GiudGiud** [reply]
ah ok. Then we dont have it.

I think it would be the MaterialArrayAux and you'd have to create it

**Reply by jessecarterMOOSE** [reply]
Is there already an aux kernel base class that I could inherit from that can hold arbitrary size mat props?
